components/script/dom/html/htmlformelement.rs

```diff
@@ -1060,6 +1061,17 @@ impl HTMLFormElement {
         load_data.referrer = referrer;
         load_data.referrer_policy = referrer_policy;
 
+        // Note the pending form navigation if this is an iframe;
+        // necessary for deciding whether to run the iframe load event steps.
+        if let Some(window_proxy) = target.undiscarded_window_proxy() {
```

Comment:
can you please confirm my understanding of this code, we mark the frame as having a pending navigation so that its about:blank load doesn't fire ?  Without this, we'dd dispatch load for about:blank and then again for the real navigation, which is not what browsers do and causes Servo issue??

Reply:
Yes that's basically it. When a navigation has started, script expects the load event to fire on the result of the navigation, but since the blank doc could still be pending it's own async load event, this flag ensure those do not fire. If the load event had fired synchronously, this wouldn't be a problem because it would have fired already before script could navigate it.